docs/azure/sdk/thread-safety.md

```diff
@@ -0,0 +1,122 @@
+
+All Azure SDK client objects are thread-safe and independent of each other. This design ensures that reusing client instances is always safe, even across threads. For example, the following code launches multiple tasks but is thread safe:
+
+```csharp
+var client = new SecretClient(
+    new Uri("<secrets_endpoint>"), new DefaultAzureCredential());
+
+foreach (var secretName in secretNames)
+{
+    // Using clients from parallel threads
+    Task.Run(() => Console.WriteLine(client.GetSecret(secretName).Value));
+}
+```
+
+Model objects used by SDK clients, whether input or output models, are *not* thread-safe by default. Most use cases involving model objects only involve a single thread, making the cost of implementing synchronization as a default behavior too high for these objects. The following sample illustrates a bug where accessing a model from multiple threads could cause an undefined behavior.
+
+```csharp
+KeyVaultSecret newSecret = client.SetSecret("secret", "value");
+
+foreach (var tag in tags)
+{
+    // Don't use model type from parallel threads
+    Task.Run(() => newSecret.Properties.Tags[tag] = CalculateTagValue(tag));
+}
+
+client.UpdateSecretProperties(newSecret.Properties);
+```
+
+To access the model from different threads, you must implement your own synchronization code. For example:
+
+```csharp
+KeyVaultSecret newSecret = client.SetSecret("secret", "value");
+
+foreach (var tag in tags)
+{
+    Task.Run(() =>
+    {
+        lock (newSecret)
+        {
+            newSecret.Properties.Tags[tag] = CalculateTagValue(tag);
+        }
+    );
+}
+
+client.UpdateSecretProperties(newSecret.Properties);
+```
+
+## Client lifetime
+
+Because Azure SDK clients are thread-safe, there's no reason to construct multiple SDK client objects for a given set of constructor parameters. Therefore, you should treat Azure SDK client objects as singletons once constructed. This is most commonly implemented by registering Azure SDK client objects as singletons in the app's Inversion of Control (IoC) container and using dependency injection (DI) to obtain references to the SDK client object. The following example shows how this is done in an ASP.NET Core app with the built-in IoC container.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddControllersWithViews();
+
+    var blobServiceClient = new BlobServiceClient(
+        new Uri("<secrets_endpoint>"), new DefaultAzureCredential());
+    services.AddSingleton(blobServiceClient);
+          
+    // Code omitted for brevity
+}
+```
+
+For more information about implementing DI with the Azure SDK, see [Dependency injection with the Azure .NET SDK](./dependency-injection.md).
+
+Alternatively, you may also create an SDK client instance and pass it as a parameter to methods that require a client. The point is to avoid unnecessary instantiations of the same SDK client object with the same parameters, as it's unnecessary and wasteful.
```

Comment:
```suggestion
Alternatively, you may create an SDK client instance and provide it to methods that require a client. The point is to avoid unnecessary instantiations of the same SDK client object with the same parameters. It's both unnecessary and wasteful.
```